pkg/containerutil/cp_linux.go

```diff
@@ -19,29 +19,66 @@ package containerutil
 import (
 	"context"
 	"fmt"
+	"io/fs"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strconv"
 	"strings"
 
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/mount"
 	"github.com/containerd/nerdctl/pkg/rootlessutil"
 	"github.com/containerd/nerdctl/pkg/tarutil"
 	securejoin "github.com/cyphar/filepath-securejoin"
 	"github.com/sirupsen/logrus"
 )
 
 // CopyFiles implements `nerdctl cp`.
-//
 // See https://docs.docker.com/engine/reference/commandline/cp/ for the specification.
-func CopyFiles(ctx context.Context, container2host bool, pid int, dst, src string, followSymlink bool) error {
+func CopyFiles(ctx context.Context, client *containerd.Client, container containerd.Container, container2host bool, dst, src string, snapshotter string, followSymlink bool) error {
 	tarBinary, isGNUTar, err := tarutil.FindTarBinary()
 	if err != nil {
 		return err
 	}
 	logrus.Debugf("Detected tar binary %q (GNU=%v)", tarBinary, isGNUTar)
-	var srcFull, dstFull string
-	root := fmt.Sprintf("/proc/%d/root", pid)
+	var srcFull, dstFull, root string
+	var cleanup func()
+	task, err := container.Task(ctx, nil)
+	if err != nil {
+		// if the task is simply not found, we should try to mount the snapshot. any other type of error from Task() is fatal here.
+		// Rootless does not support copying out of stopped/created containers
```

Comment:
Could you add the reason?

Reply:
Added a comment explaining. Lmk if I need to additional details.